jdaviz/core/template_mixin.py

```diff
@@ -68,3 +76,219 @@ def _mxn_update_plugin_opened(self, new_value):
         app_state = self.app.state
         tray_names_open = [app_state.tray_items[i]['name'] for i in app_state.tray_items_open]
         self.plugin_opened = app_state.drawer and self._registry_name in tray_names_open
+
+
+class BasePluginComponent(HubListener):
+    """
+    This Base class handles attaching traitlets from the plugin itself to logic
```

Comment:
```suggestion
    This base class handles attaching traitlets from the plugin itself to logic
```

Reply:
If this is eventually going to be the base class for all plugin components, it needs to be better documented.

I almost want to suggest abstract/meta class but maybe that is overengineering.